File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char,int> count;
        int low=0;
        int max_freq=0;
        int result=0;
        for(int high=0;high<s.length();high++){
            count[s[high]]++;
            max_freq=max(max_freq,count[s[high]]);

            while((high-low+1)-max_freq>k){
                count[s[low]]--;
                low++;
            }
            result=max(result,high-low+1);
        }
        return result;
        
    }
};

File: Data Structures & Algorithms/permutation-string/submission-1.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        if(len(s1)>len(s2)):
            return False
        l1=[0 for _ in range(26)]
        l2=[0 for _ in range(26)]
        for i in range(len(s1)):
            l1[ord(s1[i])-ord('a')]+=1
            l2[ord(s2[i])-ord('a')]+=1
        count=0
        for i in range(len(l1)):
            if l1[i]==l2[i]:
                count+=l1[i]
        if count==len(s1):
            return True
        elif len(s1)==len(s2):
            return False
        l2[ord(s2[0])-ord('a')]-=1
        low=1
        high=len(s1)
        while(high<len(s2)):
            count=0
            l2[ord(s2[high])-ord('a')]+=1
            for i in range(len(l1)):
                if l1[i]==l2[i]:
                    count+=l1[i]
            if count==len(s1):
                return True
            else:
                l2[ord(s2[low])-ord('a')]-=1
                low+=1;
                high+=1;
        return False


